assembl/models/discussion.py

```diff
@@ -561,19 +569,25 @@ def decorate_instance(
                     instance.discussion_id = parent_instance.id
 
             def contains(self, parent_instance, instance):
-                return True
+                from ..auth.util import get_current_user_id
+                try:
+                    current_user = get_current_user_id()
+                    # shortcut
+                    if instance.id == current_user:
+                        return True
+                except RuntimeError:
+                    pass
+                participants = parent_instance.get_participants_query(True)
+                return parent_instance.db.query(
+                    literal(instance.id).in_(participants.subquery())).first()[0]
 
             def get_instance(self, key, parent_instance):
                 if key == 'current':
-                    from pyramid.threadlocal import get_current_request
-                    from pyramid.httpexceptions import HTTPUnauthorized
-                    request = get_current_request()
-                    if request is not None:
-                        key = request.authenticated_userid
-                        if key is None:
-                            raise HTTPUnauthorized()
-                    else:
-                        raise RuntimeError()
+                    from ..auth.util import get_current_user_id
+                    try:
+                        key = get_current_user_id()
+                    except RuntimeError:
+                        raise HTTPUnauthorized()
```

Comment:
No this is in case of actual error (calling from outside a request, for example.)
If not logged in, get_current_user_id will just return None, but the call will work.